File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        li=[]
        for i in s:
            if i in ['[','{','(']:
                li.append(i)
            else:
                if len(li)==0:
                    return False
                else:
                    t=li[len(li)-1]
                    if t=='{' and i=='}' or t=='[' and i==']' or t=='(' and i==')':
                        li.pop(len(li)-1)
                    else:
                        return False
        return len(li)==0

        
        

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character>sk=new Stack<>();
        for(int i=0;i<s.length();i++)
        {
            char a=s.charAt(i);
            if(a=='['||a=='('||a=='{')
            {
                sk.push(a);
            }
            else
            {
                if(sk.isEmpty())
                {
                    return false;
                }
                else
                {
                    char t=sk.peek();
                    if( t=='[' && a==']' || t=='{' && a=='}' || t=='(' && a==')')
                    {
                        sk.pop();
                    }
                    else{return false;}
                }
            }
        }
        return sk.isEmpty();
    }
}